plugins/Files/js/sagas/files.js

```diff
@@ -183,15 +206,27 @@ function* getContractCountSaga() {
 
 function* renameFileSaga(action) {
 	try {
-		yield siadCall({
-			url: '/renter/rename/' + action.siapath,
-			method: 'POST',
-			qs: {
-				newsiapath: action.newsiapath,
-			},
-		})
-		yield put(actions.getFiles())
-		yield put(actions.hideRenameDialog())
+		if (action.file.type === 'file') {
+			yield siadCall({
+				url: '/renter/rename/' + action.file.siapath,
+				method: 'POST',
+				qs: {
+					newsiapath: action.newsiapath,
+				},
+			})
+			yield put(actions.getFiles())
+		}
+		if (action.file.type === 'directory') {
+			const directorypath = action.file.siapath
+			const response = yield siadCall('/renter/files')
+			const siafiles = ls(List(response.files), directorypath)
+			for (const i in siafiles.toArray()) {
+				const file = siafiles.get(i)
+				const newfilepath = Path.join(action.newsiapath, file.siapath.split(directorypath)[1])
```

Comment:
in this case yes, because each file in `siafiles` is guaranteed to have `directorypath` in its siapth (that's how `ls` works)